host/sgx/create.c

```diff
@@ -655,7 +789,8 @@ oe_result_t oe_sgx_build_enclave(
         image_size, &props, &enclave_end, &enclave_size));
 
     /* Perform the ECREATE operation */
-    OE_CHECK(oe_sgx_create_enclave(context, enclave_size, &enclave_addr));
+    OE_CHECK(
```

Comment:
If `enclave_size` is written to the base image, then the base image hash changes, so we can't do that at the time `_calculate_enclave_size()` is currently used. I'll take another look, but I don't think it's possible to get rid of the second size computation as we do actually need two different sizes in different parts of the code.
The `enclave_end` looked strange to me too, but the numbers matched up. That may have been an accident though :-)

Reply:
You were right, it works for `enclave_size` (see new code), but it doesn't work for `enclave_end` because that's patched into the image. I could calculate the EEID offset by other means, but I think it's best to leave the call to `_calculate_enclave_size()` in there, to avoid duplicating the code for size computation in other places.